src/payment/checkout/payment-form/StripePaymentForm.test.jsx

```diff
@@ -0,0 +1,283 @@
+                  isBulkOrder
+                  handleSubmit={() => {}}
+                  onSubmitPayment={() => {}}
+                  onSubmitButtonClick={() => {}}
+                />
+              </Elements>
+            </reactRedux.Provider>
+          </AppContext.Provider>
+        </IntlProvider>
+      ));
+
+      const formValues = {
+        firstName: '',
+        lastName: '',
+        address: '',
+        city: '',
+        country: 'UK',
+        cardNumber: '',
+        securityCode: '',
+        cardExpirationMonth: '',
+        cardExpirationYear: '',
+        optionalField: '',
+        organization: 'edx',
+      };
+
+      const requiredFields = formValidators.getRequiredFields(formValues, isBulkOrder);
+      expect(formValues.organization).toEqual(requiredFields.organization);
+    });
+  });
+  describe('onSubmit', () => {
+    it('throws expected errors', () => {
+      state = {
+        payment: {
+          basket: {
+            loading: true,
+            loaded: false,
+            submitting: false,
+            redirect: false,
+            isBasketProcessing: false,
+            products: [{ sku: '00000' }],
+            enableStripePaymentProcessor: true,
+          },
+          clientSecret: { isClientSecretProcessing: false, clientSecretId: '' },
+        },
+        feedback: { byId: {}, orderedIds: [] },
+        form: {
+          payment: {
+            submitErrors: { firstName: 'error' },
+          },
+        },
+      };
+      const submitStripePayment = jest.fn();
+      mount((
+        <IntlProvider locale="en">
+          <AppContext.Provider value={{ authenticatedUser }}>
+            <reactRedux.Provider store={mockStore(state)}>
+              <Elements stripe={mockStripe}>
+                <StripePaymentForm
+                  handleSubmit={() => {}}
+                  onSubmitPayment={() => submitStripePayment}
+                  onSubmitButtonClick={() => {}}
+                  shouldFocusFirstError
+                  firstErrorId={null}
+                />
+              </Elements>
+            </reactRedux.Provider>
+          </AppContext.Provider>
+        </IntlProvider>
+      ));
+      const SubmissionError = jest.fn();
+      const testData = [
+        [
+          { firstName: 'This field is required' },
+          new SubmissionError({
+            firstName: 'This field is required',
+          }),
+        ],
+        [
+          {},
+          null,
+        ],
+      ];
+
+      testData.forEach((testCaseData) => {
+        validateRequiredFieldsMock.mockReturnValueOnce(testCaseData[0]);
+        if (testCaseData[1]) {
+          expect(() => submitButton.simulate('click'));
+          expect(submitStripePayment).not.toHaveBeenCalled();
+        } else {
+          expect(() => submitButton.simulate('click')).not.toThrow();
+        }
+      });
+    });
+  });
+
+  describe('validateRequiredFields', () => {
+    it('returns errors if values are empty', () => {
```

Comment:
Is it worth testing for Stripe if the required fields for the certain countries is being required onSubmit?

Reply:
like state and zip? good call Ive added those into the getRequiredFields

Reply:
Great, thanks! One thing to note is that in one of the example `testFormValues`, UK should have postal code since it's required in the Stripe implementation.

I see these examples were taken from the `PaymentForm` tests, but would be good to have this for Stripe with an example of a country where state and postal code is required (aka. US), where only postal code is required (UK) and where only state is required (India)?

Reply:
Added